regression/library/test_vcache_flush.cpp

```diff
@@ -169,15 +190,26 @@ int test_vcache_flush() {
 
         addr_bitwidth = hb_mc_config_get_network_bitwidth_addr(config);
         addrs[0] = 0;
-        addrs[1] = 1 << (addr_bitwidth-2); // Convert to word-level addr
+        addrs[NUM_TESTS - 1] = (1 << (addr_bitwidth-1)) - 1; // Word-level addr of highest word
+
+        // Generate some random addresses
+        for(int i = 1; i < NUM_TESTS - 1; i++)
+                addrs[i] = rand() % addrs[NUM_TESTS - 1];
 
-        data = rand();
+        uint32_t data32 = rand();
+        uint16_t data16 = rand() & 0xFFFF;
+        uint8_t   data8 = rand() & 0xFF;
 
-        for (dram = 0; dram < ndrams; ++dram){
+        for (dram = 0; dram < ndrams; ++dram) {
                 dram_coord_x = dram_xs[dram];
-                for(i = 0; i < NUM_TESTS; ++i){
+                for(i = 0; i < NUM_TESTS; ++i) {
                         addr = addrs[i];
```

Comment:
Suggestion; generate the random data here so that it is not reused between tests.